src/components/delete-with-confirmation-icon.tsx

```diff
@@ -0,0 +1,39 @@
+import { Box, Button, IconButton } from '@mui/material';
+import React, { useState } from 'react';
+import { useTranslator } from '../hooks';
+import CloseIcon from '@mui/icons-material/Close';
+
+export function DeleteWithConfirmationIcon(props: {
+  name: string | undefined;
+  clickHandler: () => void;
+}): JSX.Element | null {
+  const [clicked, setClicked] = useState(false);
+
+  const trans = useTranslator('jupyterlab');
+
+  const buttonTitle = props.name
+    ? trans.__('Delete "%1"', props.name)
+    : trans.__('Delete job');
+
+  return (
+    <Box sx={{ width: '6em' }}>
+      {clicked ? (
+        <Button
+          variant="contained"
+          color="error"
+          title={buttonTitle}
+          onClick={props.clickHandler}
+          onBlur={_ => setClicked(false)}
+          style={{ visibility: clicked ? 'visible' : 'hidden' }}
+          autoFocus
+        >
+          {trans.__('Delete')}
+        </Button>
+      ) : (
+        <IconButton title={buttonTitle} onClick={_ => setClicked(true)}>
```

Comment:
I think this is where you would want to try `stopPropagation` and/or `preventDefault` to fix the bug where another row shows the red Delete button after the first is deleted.

Reply:
Fixed this. The problem was that in unique `key` prop was not passed to `TableRow` or `TableCell`, mass-rendered components in `job-row.tsx` and `job-definition-row.tsx`. Passing unique key to either `TableRow` or `TableCell` solves the problem. I opted for passing a unique key to both